sdcm/nemesis.py

```diff
@@ -671,7 +671,41 @@ def do_refresh(node):
             node.remoter.sudo(f'rm -f /var/lib/scylla/data/keyspace1/{upload_dir}/upload/schema.cql')
             node.remoter.sudo(f'rm -f /var/lib/scylla/data/keyspace1/{upload_dir}/upload/manifest.json')
             self.log.debug(f'Loading {keys_num} keys to {node.name} by refresh')
+
+            # Resharding of the loaded sstable files is performed before they are moved from upload to the main folder.
+            # So we need to validate that resharded files are placed in the "upload" folder before moving.
+            # Find the compaction output that reported about the resharding
+
+            search_reshard = node.follow_system_log(patterns=['Resharded.*\[/'])
             node.run_nodetool(sub_cmd="refresh", args="-- keyspace1 standard1")
+            resharding_done = list(search_reshard)
+            return resharding_done
+
+        def validate_resharding_after_refresh(resharding_done):
+            """
+            # Validate that files after resharding were saved in the "upload" folder.
+            # Example of compaction output:
+
+            #   scylla[6653]:  [shard 0] compaction - [Reshard keyspace1.standard1 3cad4140-f8c3-11ea-acb1-000000000002]
+            #   Resharded 1 sstables to [
+            #   /var/lib/scylla/data/keyspace1/standard1-9fbed8d0f8c211ea9bb1000000000000/upload/md-9-big-Data.db:level=0,
+            #   /var/lib/scylla/data/keyspace1/standard1-9fbed8d0f8c211ea9bb1000000000000/upload/md-10-big-Data.db:level=0,
+            #   /var/lib/scylla/data/keyspace1/standard1-9fbed8d0f8c211ea9bb1000000000000/upload/md-11-big-Data.db:level=0,
+            #   /var/lib/scylla/data/keyspace1/standard1-9fbed8d0f8c211ea9bb1000000000000/upload/md-12-big-Data.db:level=0,
+            #   /var/lib/scylla/data/keyspace1/standard1-9fbed8d0f8c211ea9bb1000000000000/upload/md-13-big-Data.db:level=0,
+            #   /var/lib/scylla/data/keyspace1/standard1-9fbed8d0f8c211ea9bb1000000000000/upload/md-22-big-Data.db:level=0,
+            #   /var/lib/scylla/data/keyspace1/standard1-9fbed8d0f8c211ea9bb1000000000000/upload/md-15-big-Data.db:level=0,
+            #   /var/lib/scylla/data/keyspace1/standard1-9fbed8d0f8c211ea9bb1000000000000/upload/md-16-big-Data.db:level=0,
+            #   ]. 91MB to 92MB (~100% of original) in 5009ms = 18MB/s. ~370176 total partitions merged to 370150
+            """
```

Comment:
@juliayakovlev just to understand how it works.
Do the files remain in the upload folder after the refresh command finishes?

Reply:
@roydahan
No. The files will be copied to the main folder. There is data validation in the nemesis after "refresh" is completed.

Reply:
Copied or moved?
You are verifying here that the sstables have the upload path in their name so I don't understand....

Reply:
@roydahan

Moved, sorry.

The procedure is:
1. put the db files for refresh in the "upload" folder
2. run "nodetool refresh"
3. refresh command will start resharding
4. files after resharding are placed into "upload" folder
5. move resharded files from "upload" to main folder

So, I am checking that resharded files are in upload folder, not in the main

Reply:
Ok, now that make sense :)
Worth having it in the commit message.

Reply:
Done